src/symbols/Edge.tsx

```diff
@@ -2,19 +2,34 @@ import React, { FC, useMemo } from 'react';
 import { useSpring, a } from '@react-spring/three';
 import { Arrow } from './Arrow';
 import { Label } from './Label';
-import { animationConfig, getMidPoint, getPoints } from '../utils';
+import {
+  animationConfig,
+  getLabelOffsetByType,
+  getMidPoint,
+  getPoints
+} from '../utils';
 import { Line } from './Line';
-import { Theme } from '../utils/themes';
-import { InternalGraphEdge } from '../types';
+import { Theme } from '../utils';
 import { useStore } from '../store';
+import { Euler } from 'three';
+
+const LABEL_FONT_SIZE = 6;
+
+export type EdgeLabelPosition = 'inside' | 'outside' | 'inline';
 
 export interface EdgeProps {
   id: string;
   theme: Theme;
   animated?: boolean;
+  labelPlace?: EdgeLabelPosition;
 }
 
-export const Edge: FC<EdgeProps> = ({ id, animated, theme }) => {
+export const Edge: FC<EdgeProps> = ({
+  id,
+  animated,
+  theme,
+  labelPlace = 'inline'
```

Comment:
put default initializers in the `defaultProps` below. Long story around this but its safer.